### PR TITLE
Change aggregate_values output to remain vectorized

### PR DESCRIPTION
**Pull request by bmcase**
This small PR is to change the output type of aggregate_value to still have it in vectorized form.   This will let the aggregated result flow vectorized into the DP noise stage (future PR will add).  It also lets the `aggregate_values` function be used in a more standalone way to aggregate Bernoulli noise to form Binomial samples.

The changes are to: 
1. the `aggregate_values` function and its tests (all of this is complete except the last property tests which are currently commented out with todo!()).  
2. where `aggregate_values` is used in the aggregation stage (`aggregate_contributions`)
3. where `aggregate_values` is used in `compute_feature_label_dot_product` 

Currently in 2. and 3. these functions now carry out the transpose operation to come out of vectorized form.  With the DP noise PR, we will further change this so that `aggregate_contributions` and `compute_feature_label_dot_product` both continue to have outputs which are vectorized to flow into DP noise which will then transform it back at the end.

### CODE REVIEW

File: ipa-core/src/protocol/ipa_prf/aggregation/mod.rs [hunk 1]
```diff
@@ -372,17 +372,17 @@ pub mod tests {
...
                     let num_rows = inputs.len();
                     aggregate_values::<BA8, 8>(ctx, stream::iter(inputs).boxed(), num_rows)
                 })
                 .await
                 .map(Result::unwrap)
-                .reconstruct();
-
-            assert_eq!(result, [0u32, 1, 2, 3, 4, 5, 6, 7].map(BA8::truncate_from));
+                .reconstruct_arr();
+            let expected: BitDecomposed<BA8> = input_row(8, &[0u32, 1, 2, 3, 4, 5, 6, 7])
```
Comment (benjaminsavage) [inline ipa-core/src/protocol/ipa_prf/aggregation/mod.rs:383]: Wait, shouldn't we expect a transposed result now

File: ipa-core/src/protocol/ipa_prf/aggregation/mod.rs [hunk 2]
```diff
@@ -395,17 +395,20 @@ pub mod tests {
...
                 })
                 .await
                 .map(Result::unwrap)
-                .reconstruct();
+                .reconstruct_arr();
 
-            assert_eq!(result, [0u32, 1, 2, 3, 0, 0, 0, 0].map(BA8::truncate_from));
+            assert_eq!(
+                result,
+                input_row(8, &[0u32, 1, 2, 3, 0, 0, 0, 0])
```
Comment (benjaminsavage) [inline ipa-core/src/protocol/ipa_prf/aggregation/mod.rs:409]: I'm surprised you're still expecting the same thing given this method now returns a new result.

**Review by benjaminsavage — COMMENTED** [review]
We discussed this PR over VC and I agree this is the right way to go. Once the tests are all passing let's merge.

File: ipa-core/src/protocol/ipa_prf/aggregation/mod.rs (hunk 1 above)
Comment (bmcase) [reply ipa-core/src/protocol/ipa_prf/aggregation/mod.rs:383]: yes, so here `input_row` applies transpose (turning into vectorized form) to the expected result.  So the `result` is still in vectorized from when we compare it in the `assert_eq`

File: ipa-core/src/protocol/ipa_prf/aggregation/mod.rs (hunk 2 above)
Comment (bmcase) [reply ipa-core/src/protocol/ipa_prf/aggregation/mod.rs:409]: similar comment thread above, but here `result` is vectorized still and `input_row` is vectorizing the expected result.  Previously `input_row` wouldn't have appeared here.

**Comment by bmcase** [comment]
Added @andyleiserson 's fix for property tests and those are working now.  Working on getting the rest of the checks to pass now.

**Comment by bmcase** [comment]
some unrelated tests are failing. I recently rebased so not sure why that would be.

**Comment by benjaminsavage** [comment]
I don't think these are unrelated tests. They are the tests relating to aggregation and the IPA protocol itself. Happy to merge this once the tests are all passing.

**Pull request merged**